symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java

```diff
@@ -39,57 +26,47 @@
 @API(status = API.Status.EXPERIMENTAL)
 public class SymphonyBdk {
 
-  private final ApiClientFactory apiClientFactory;
-  private final ApiClient podClient;
-  private final ApiClient agentClient;
-
-  private final AuthSession botSession;
   private final OboAuthenticator oboAuthenticator;
 
-  private final DatafeedService datafeedService;
-  private final UserService userService;
-  private final StreamService streamService;
-  private final SessionService sessionService;
   private final ActivityRegistry activityRegistry;
+  private final SessionService sessionService;
+  private final StreamService streamService;
+  private final UserService userService;
+  private final MessageService messageService;
+  private final DatafeedService datafeedService;
 
   public SymphonyBdk(BdkConfig config) throws AuthInitializationException, AuthUnauthorizedException {
-
-    this.apiClientFactory = new ApiClientFactory(config);
-    this.podClient = apiClientFactory.getPodClient();
-    this.agentClient = apiClientFactory.getAgentClient();
-
-    final AuthenticatorFactory authenticatorFactory = new AuthenticatorFactory(config, this.apiClientFactory);
-
-    this.botSession = authenticatorFactory.getBotAuthenticator().authenticateBot();
+    ApiClientFactory apiClientFactory = new ApiClientFactory(config);
+    final AuthenticatorFactory authenticatorFactory = new AuthenticatorFactory(config, apiClientFactory);
+    AuthSession botSession = authenticatorFactory.getBotAuthenticator().authenticateBot();
     this.oboAuthenticator = config.isOboConfigured() ? authenticatorFactory.getOboAuthenticator() : null;
-
-    // setup the datafeed
-    final DatafeedApi datafeedApi = new DatafeedApi(this.agentClient);
-    if (DatafeedVersion.of(config.getDatafeed().getVersion()) == DatafeedVersion.V2) {
-      this.datafeedService = new DatafeedServiceV2(datafeedApi, this.botSession, config);
-    } else {
-      this.datafeedService = new DatafeedServiceV1(datafeedApi, this.botSession, config);
-    }
-    // setup other services
-    this.userService = new UserService(new UserApi(this.podClient), new UsersApi(this.podClient), this.botSession);
-    this.streamService = new StreamService(new StreamsApi(this.podClient), this.botSession);
-    this.sessionService = new SessionService(new SessionApi(this.podClient));
-    this.activityRegistry = new ActivityRegistry(this.sessionService.getSession(this.botSession), this.datafeedService::subscribe);
+    ServiceFactory serviceFactory =
+        new ServiceFactory(apiClientFactory.getPodClient(), apiClientFactory.getAgentClient(), botSession, config);
+
+    // service init
+    this.sessionService = serviceFactory.getSessionService();
+    this.userService = serviceFactory.getUserService();
+    this.streamService = serviceFactory.getStreamService();
+    this.messageService = serviceFactory.getMessageService();
+    this.datafeedService = serviceFactory.getDatafeedService();
```

Comment:
Then the `serviceFactory` could return new instances, not singleton instances

Reply:
yep